bound/kt/src/main/kotlin/tbdex/sdk/resources/Offering.kt

```diff
@@ -0,0 +1,218 @@
+        this.data = OfferingData.fromRustCore(rustCoreOffering.getData().data)
+        this.signature = rustCoreOffering.getData().signature
+    }
+
+    constructor(json: String) {
+        this.rustCoreOffering = RustCoreOffering.fromJsonString(json)
+
+        this.metadata = rustCoreOffering.getData().metadata
+        this.data = OfferingData.fromRustCore(rustCoreOffering.getData().data)
+        this.signature = rustCoreOffering.getData().signature
+    }
+
+    constructor(rustCoreOffering: RustCoreOffering) {
+        this.rustCoreOffering = rustCoreOffering
+
+        this.metadata = this.rustCoreOffering.getData().metadata
+        this.data = OfferingData.fromRustCore(this.rustCoreOffering.getData().data)
+        this.signature = this.rustCoreOffering.getData().signature
+    }
+
+    fun toJson(): String {
+        return this.rustCoreOffering.toJson()
+    }
+}
+
+data class OfferingData(
+    val description: String,
+    val payoutUnitsPerPayinUnit: String,
+    val payin: PayinDetails,
+    val payout: PayoutDetails,
+    val requiredClaims: PresentationDefinition?
+) {
+    fun toRustCore(): RustCoreOfferingData {
+        return RustCoreOfferingData(
+            this.description,
+            this.payoutUnitsPerPayinUnit,
+            this.payin.toRustCore(),
+            this.payout.toRustCore(),
+            this.requiredClaims?.rustCorePresentationDefinition?.getData()
+        )
+    }
+
+    companion object {
+        internal fun fromRustCore (rustCoreOfferingData: RustCoreOfferingData): OfferingData {
+            return OfferingData(
+                rustCoreOfferingData.description,
+                rustCoreOfferingData.payoutUnitsPerPayinUnit,
+                PayinDetails.fromRustCore(rustCoreOfferingData.payin),
+                PayoutDetails.fromRustCore(rustCoreOfferingData.payout),
+                rustCoreOfferingData.requiredClaims?.let { PresentationDefinition(it) }
+            )
+        }
+    }
+}
+
+data class PayinDetails(
+    val currencyCode: String,
+    val min: String? = null,
+    val max: String? = null,
+    val methods: List<PayinMethod>
+) {
+    fun toRustCore(): RustCorePayinDetails {
+        return RustCorePayinDetails(
+            this.currencyCode,
+            this.min,
+            this.max,
+            this.methods.map { it.toRustCore() }
+        )
+    }
+
+    companion object {
+        internal fun fromRustCore(rustCorePayinDetails: RustCorePayinDetails): PayinDetails {
+            return PayinDetails(
+                rustCorePayinDetails.currencyCode,
+                rustCorePayinDetails.min,
+                rustCorePayinDetails.max,
+                rustCorePayinDetails.methods.map { PayinMethod.fromRustCore(it) }
+            )
+        }
+    }
+}
+
+data class PayinMethod(
+    val kind: String,
+    val name: String? = null,
+    val description: String? = null,
+    val group: String? = null,
+    val requiredPaymentDetails: JsonNode? = null,
```

Comment:
Having JsonNode here is a bit sus right?

could name these `val jsonRequiredPaymentDetails: String` or something?

Reply:
nope this is right, here's the existing https://github.com/TBD54566975/tbdex-kt/blob/8c6a93b47b1cd0eefdc5910b26c04319e54ad970/protocol/src/main/kotlin/tbdex/sdk/protocol/models/ResourceData.kt#L142

we stringify things over the UniFFI boundary, which is why this (Offering) and the Rfq (both of which have JSON types) are fully written out in kotlin as opposed to using `typealias Offering = RustCoreOfferingData`, and then you see below the `toRustCore()` and `fromRustCore()` which stringify/parse over the UniFFI boundary